src/main/java/com/spikes2212/control/PIDSparkMax.java

```diff
@@ -0,0 +1,137 @@
+package com.spikes2212.control;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANPIDController;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.ControlType;
+
+import java.util.function.Supplier;
+
+/**
+ * A {@link CANSparkMax} that can run a PID loop.
+ *
+ * @author Eran Goldstein
+ */
+public class PIDSparkMax implements PIDSpeedController {
+
+    /**
+     * The {@link CANSparkMax} on which the PID is calculated.
+     */
+    private final CANSparkMax sparkMax;
+
+    /**
+     * The {@link CANPIDController} which is responsible for handling the PID loop.
+     */
+    private final CANPIDController pidController;
+
+    /**
+     * The {@link CANEncoder} of the motor connected to the spark max.
+     */
+    private final CANEncoder encoder;
+
+    /**
+     * The PID loop's {@link PIDSettings}.
+     */
+    private final PIDSettings settings;
+
+    /**
+     * The PID loop's feed forward constant.
+     */
+    private final Supplier<Double> kF;
+
+    /**
+     * The PID loop's {@link ControlType}.
+     */
+    private final ControlType mode;
+
+    /**
+     * Constructs a PIDSparkMax instance with the given parameters as field values.
+     *
+     * @param sparkMax The Spark Max speed controller on which the PID loop is calculated.
+     * @param settings The PID loop's {@link PIDSettings}.
+     * @param kF       The PID loop's feed forward constant.
+     * @param mode     The PID loop's {@link ControlMode}.
+     */
+    public PIDSparkMax(CANSparkMax sparkMax, PIDSettings settings, Supplier<Double> kF, ControlType mode, int timeout) {
+        this.sparkMax = sparkMax;
+        this.pidController = sparkMax.getPIDController();
+        this.encoder = sparkMax.getEncoder();
+        this.settings = settings;
+        this.kF = kF;
+        this.mode = mode;
+        sparkMax.setCANTimeout(timeout);
+    }
+
+    @Override
+    public void configureLoop(Supplier<Double> maxSpeed, Supplier<Double> minSpeed) {
+        sparkMax.restoreFactoryDefaults();
+
+        pidController.setOutputRange(maxSpeed.get(), minSpeed.get());
```

Comment:
switch between the parameter's positions. setOutputRange gets the minimum speed before the maximum speed